napalm_panos/panos.py

```diff
@@ -31,11 +31,12 @@
 
 from napalm_base.base import NetworkDriver
 from netmiko import ConnectHandler
+from netmiko import __version__ as netmiko_version
 
 
 class PANOSDriver(NetworkDriver):
 
-    def __init__(self, hostname, username, password, timeout=60, optional_args=None):
+    def __init__(self, hostname, username, password='', timeout=60, optional_args=None):
```

Comment:
You could write code that was incompatible; but you could also write code that is compatible. The arguments become optional, but they are still supported by all platforms. If you want to pass fictional username/password arguments (for compatibility), then you could.

You would already have to construct the optional_args per device to pass the SSH keys.